packages/stark-core/src/modules/user/user.module.ts

```diff
@@ -8,14 +9,16 @@ export class StarkUserModule {
 	 * Instantiates the services only once since they should be singletons
 	 * so the forRoot() should be called only by the AppModule
 	 * @link https://angular.io/guide/singleton-services#forroot
+	 * @param userModuleConfig - Object containing the configuration (if any) for the User Module
 	 * @returns a module with providers
 	 */
-	public static forRoot(): ModuleWithProviders {
+	public static forRoot(userModuleConfig?: StarkUserModuleConfig): ModuleWithProviders {
 		return {
 			ngModule: StarkUserModule,
 			providers: [
 				{ provide: STARK_USER_SERVICE, useClass: StarkUserServiceImpl },
-				{ provide: STARK_USER_REPOSITORY, useClass: StarkUserRepositoryImpl }
+				{ provide: STARK_USER_REPOSITORY, useClass: StarkUserRepositoryImpl },
+				userModuleConfig ? { provide: STARK_USER_PROFILE_RESOURCE_PATH, useValue: userModuleConfig.userProfileResourcePath } : []
```

Comment:
This code seems weird to me...
Should not it be: 

```typescript
providers: [
  { provide: STARK_USER_SERVICE, useClass: StarkUserServiceImpl },
  { provide: STARK_USER_REPOSITORY, useClass: StarkUserRepositoryImpl },
  ... userModuleConfig ? { provide: STARK_USER_PROFILE_RESOURCE_PATH, useValue: userModuleConfig.userProfileResourcePath } : []
]
```

instead ? :blush:

Otherwise, if there is no `userModuleConfig` you will have the following result object which contains an empty array : 
```typescript
providers: [
  { provide: STARK_USER_SERVICE, useClass: StarkUserServiceImpl },
  { provide: STARK_USER_REPOSITORY, useClass: StarkUserRepositoryImpl },
  []
]
```

Reply:
About your suggestion... I don't understand... what should be changed?

In fact, `[]` is a valid value to pass to module `providers`... this is the typings for `Provider`:

```typescript
export declare type Provider = TypeProvider | ValueProvider | ClassProvider | ConstructorProvider | ExistingProvider | FactoryProvider | any[];
```

Reply:
What I can do is to change it like this:

```typescript
userModuleConfig ? { provide: STARK_USER_PROFILE_RESOURCE_PATH, useValue: userModuleConfig.userProfileResourcePath } : ...[]
```

so I expand the array `...[]` insteadof passing `[]`

Reply:
I think it's weird to pass an empty array to providers. 
Your proposal seems fine since the final `providers` won't contain a useless empty array 😊

Reply:
@SuperITMan I'm afraid I cannot change the code as we want because using the spread operator `...` or any function causes the AOT compilation to fail: `Function calls are not supported in decorators but 'StarkUserModule' was called.`.

This is because the spread operator is transpiled to `Array.concat()` in ES5.... and functions cannot be called in AOT :(

So I'll leave the code as it is now, in fact we have done this in other modules... and it is the way is recommended in forums, for example: https://stackoverflow.com/a/49763160